lib/matplotlib/tests/test_constrainedlayout.py

```diff
@@ -380,6 +380,8 @@ def test_constrained_layout23():
         fig.suptitle("Suptitle{}".format(i))
 
 
+@pytest.mark.timeout(30)
+@pytest.mark.flaky(reruns=3)
```

Comment:
Can you leave a note explaining the source of randomness?
Perhaps worth investigating whether kiwisolver can be made deterministic too.

Reply:
Done.  Also opened an issue at kiwi to see if they have suggestions for making the tests more deterministic...